### DISCUSSION TITLE
Emet JS event on pagination click

### DISCUSSION THREAD

**Discussion by cth-latest** [livewire/livewire#3436] [3. Feature ideas]
Hello, I think it would be nice to natively emit a JS event when changing pages via pagination.

This allows you to load images or content that are in lazy load and need a refresh (Example https://github.com/verlok/vanilla-lazyload)

(Adding a onclick event don't work because that will reload the content before the server response)

**Comment by ph7jack** [comment]
Hello @Velka-DEV 
You can use alpinejs to do it.

```
<div class="hidden"
x-data="{ page: @entangle('page').defer }"
x-init="function() {
      $watch('page', page => {
         // Your code to starts a lazy load
      })
}"></div>
```

**Comment by joshhanley** [comment]
@Velka-DEV I think @PH7-Jack's suggestion is best are you are going to be doing other things like lazy loading images, etc.

But another option is to overload the `setPage` method from the `WithPagination` trait.

Just put this into your component

```php
public function setPage($page)
{
    $this->page = $page;
    
    $this->dispatchBrowserEvent('pageChanged');
}
```


Hope this helps!
